matchms/filtering/metadata_processing/derive_annotation_from_compound_name.py

```diff
@@ -10,16 +10,16 @@
 import pandas as pd
 import pubchempy
 from matchms import Spectrum
-from matchms.filtering.filter_utils.smile_inchi_inchikey_conversions import \
-    _check_fully_annotated
+from matchms.filtering.filter_utils.smile_inchi_inchikey_conversions import (
+    is_valid_inchi, is_valid_inchikey, is_valid_smiles)
 
 
 logger = logging.getLogger("matchms")
 
 
-def derive_smiles_from_pubchem_compound_name_search(spectrum_in: Spectrum,
-                                                    annotated_compound_names_file: Optional[str] = None,
-                                                    mass_tolerance: float = 0.1):
+def derive_annotation_from_compound_name(spectrum_in: Spectrum,
+                                         annotated_compound_names_file: Optional[str] = None,
+                                         mass_tolerance: float = 0.1):
     """Adds smiles, inchi, inchikey based on compound name by searching pubchem
 
     The smiles, inchi and inchikey are repaired if the found smiles is close enough to the parent mass.
```

Comment:
"repaired" is not how I would call this. Better describe in a bit more detail what is done by the function (including the new change that this function will only be executed if there is no valid annotation.

Reply:
Good point, this has been fixed